File: frontend/demo/component/basiclayouts/basic-layouts-horizontal-layout.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import '@vaadin/vaadin-ordered-layout/vaadin-horizontal-layout';
import './layout-item';

import '@vaadin/vaadin-item';

@customElement('basic-layouts-horizontal-layout')
export class Example extends LitElement {
  constructor() {
    super();
    this.classList.add('basic-layouts-example');
  }

  protected createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-horizontal-layout theme="spacing padding">
        <layout-item>Item 1</layout-item>
        <layout-item>Item 2</layout-item>
        <layout-item>Item 3</layout-item>
        <layout-item>Item 4</layout-item>
      </vaadin-horizontal-layout>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/basiclayouts/BasicLayoutsHorizontalLayout.java
package com.vaadin.demo.component.basiclayouts;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("basic-layouts/horizontal-layout")
public class BasicLayoutsHorizontalLayout extends Div {
    public BasicLayoutsHorizontalLayout() {
        // tag::snippet[]
        HorizontalLayout layout = new HorizontalLayout();
        layout.setPadding(true);
        
        Div item1 = new Div("Item 1");
        item1.setClassName("example-item");
        
        Div item2 = new Div("Item 2");
        item2.setClassName("example-item");
        
        Div item3 = new Div("Item 3");
        item3.setClassName("example-item");
        
        layout.add(item1, item2, item3);
        // end::snippet[]
        
        this.setClassName("basic-layouts-example");
        this.add(layout);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<BasicLayoutsHorizontalLayout> { // hidden-source-line
    } // hidden-source-line
}
